src/reference_tree.ml

```diff
@@ -155,6 +174,38 @@ let load_constraint_group_from_xml d c =
         | _ -> raise (Bad_interface_definition ("Malformed constraint: " ^ Xml.to_string c))
     in Xml.fold aux d c
 
+let load_docs_hints d c =
+    let aux d c =
+        match c with
+        | Xml.Element ("hints", attrs, [Xml.PCData s]) ->
+            let hint_type = List.assoc "type" attrs in
+            let hint = { text = s; hint_type = hint_type } in
+            let new_docs = { d.docs with hints = hint :: d.docs.hints } in
+            { d with docs = new_docs }
+        | _ -> raise (Bad_interface_definition ("Malformed hint: " ^ Xml.to_string c))
+    in aux d c
+
+let load_docs_from_xml d x =
+    let aux d x =
+        match x with
+        | Xml.Element ("headline", _, [Xml.PCData s]) -> 
+            let new_docs = {d.docs with headline = s} in 
+            {d with docs = new_docs}
+        | Xml.Element ("text", _, [Xml.PCData s]) -> 
+            let new_docs = {d.docs with text = s} in 
+            {d with docs = new_docs}
+        | Xml.Element ("hints", _, _) -> 
+            load_docs_hints d x
+        | Xml.Element ("usageExample", _, [Xml.PCData s]) -> 
+            let new_docs = {d.docs with usageExample = s} in 
+            {d with docs = new_docs}
+        | _ -> d  (* Ignore unknown elements instead of raising an error *)
+    in
+    match x with
+    | Xml.Element ("docs", _, children) ->
+        List.fold_left aux d children 
+    | _ -> d
```

Comment:
```suggestion
    in Xml.fold aux d x
```

Reply:
Granted the documentation on Xml-light is sparse (contained in its .mli files), but Xml.fold will do the left fold over children and avoid the redundancy above.